src/import/FSHImporter.ts

```diff
@@ -1509,21 +1491,34 @@ export class FSHImporter extends FSHVisitor {
             'Using "|" to list references is deprecated. Please use "or" to list references.',
             {
               file: this.currentFile,
-              location: this.extractStartStop(ctx)
+              location: this.extractStartStop(t)
             }
           );
         }
         references.forEach(r =>
-          onlyRule.types.push({
+          orTypes.push({
             type: this.aliasAwareValue(referenceToken, r),
             isReference: true
           })
         );
       } else {
-        onlyRule.types.push({ type: this.aliasAwareValue(t.name()) });
+        orTypes.push({ type: this.aliasAwareValue(t.name()) });
       }
     });
-    return onlyRule;
+
+    orTypes.forEach(orType => {
+      if (FLAGS.includes(orType.type)) {
+        logger.warn(
+          `The targetType '${orType.type}' appears to be a flag value rather than a valid target data type.`,
+          {
+            file: this.currentFile,
+            location: this.extractStartStop(ctx)
+          }
+        );
+      }
+    });
```

Comment:
I think this is more appropriate in the specific code for `AddElementRule` rather than in this shared function.  The reason for this is because there is no ambiguity w/ the grammar for `OnlyRule`s.  If the FSH says `foo only MS` then there really is no confusion -- only a type would ever go after the word `only`.  As opposed to `AddElementRule` where `* foo 0..1 MS` is a little more ambiguous since the word after `0..1` can be a _flag_ or a _type_ (although, in the context of the rest of the phrase, I think we can still figure it out).

Reply:
Good catch - fixed.